### ISSUE TITLE
Google Showtimes is closed

### ISSUE THREAD

**Issue by gendeld** [erunion/showtimes#47]
"Google Showtimes was discontinued on November 1, 2016. You can now discover showtimes on the search results page by searching for the name of the movie you want to see or try searching for the word “movies” to discover what’s in theaters."

The api is no longer working.

**Comment by DWMiller** [comment]
Yeah, isn't likely to be a fix for this one. I think the api would have to be completely redone using a difference source.

**Comment by ccinar91** [comment]
Are you planning to fix the api or is it too much of a hassle?

**Comment by erunion** [comment]
Oh geez. I hadn't heard about an announcement for this at all. 

I'll have to switch it over to different source.

**Comment by pikkle** [comment]
I guess you could try to fetch data from google still, by looking at
[https://www.google.ch/#q=movies+new+york](https://www.google.ch/#q=movies+new+york)
But it will take a few more HTTP requests it seems...

**Comment by hguiney** [comment]
Unfortunately that is considered scraping and is most likely against
Google’s terms of service.

**Comment by erunion** [comment]
@hguiney Well this library is already a scraper, so...

@pikkle Strange that that doesn't work here in the US. https://www.google.com/webhp?hl=en&ictx=2&sa=X&ved=0ahUKEwiO1sKw3JnQAhUk7IMKHRneC3UQPQgD#hl=en&q=movies+11249

**Comment by pikkle** [comment]
I haven't found a source that gives theatre showtimes outside of US/Canada other than Google movies so far... Scraping Google might be the only international free source.

@erunion You can use the 'near' keyword alongside a zipcode:
https://www.google.com/search?hl=en&ei=HEojWKzUJcvtUrKFuWA&q=movies+near+11249&oq=movies+near+11249

**Comment by erunion** [comment]
@pikkle Nice find. Might be able to use that, though pulling showtimes for a location is going to take a **lot** longer than it used to. :\

**Comment by gendeld** [comment]
The "hl=" can be used to determine the language of the results.

**Comment by erunion** [comment]
I'll see if I can get started on this this evening, so you all can get back to looking up movies ASAP.

**Comment by erunion** [comment]
Ah, the joys of managing a scraper.

**Comment by erunion** [comment]
Looks like "movie showtimes near" is the query prefix that picks up the list of movies showing in your area 100% of the time.

I can't seem to get latitude/longitude searches working on it, unfortunately, so I might need to drop support for that and only keep zipcode, address, or vague city lookups. Shame.

**Comment by erunion** [comment]
Fixing this is taking a little longer because I've decided now is the time to write in customizable engine support to support more than one service. Starting with Fandango, because the "movie showtimes near" query in Google isn't 100% consistent.

Should be done *soon*.

**Comment by alexamp** [comment]
i think Fandango works only in the US right? :\

**Comment by erunion** [comment]
@alexamp Unfortunately. At least with this new engine framework I'm working on, it'll be really easy to add support for other providers.

**Comment by i8wu** [comment]
@erunion Let me know if you need help, just came across this API. Seems like it would be great for my use case so I'd like to help get it working again.

**Comment by erunion** [comment]
@i8wu I'll see if I can wrap up the engine support I've got going this weekend. Should be a good starting point to get the library back up and running for most use-cases, and if anyone wants additional, or international, providers, those can be added fairly easily.

**Comment by gendeld** [comment]
@erunion Hey, any news about progress? any way we can help?

**Comment by erunion** [comment]
I've pushed up my, very initial, work on an engine system to the `google-movies-shutdown` branch with a work in progress PR at #51.

The branch is completely broken right now as things are still being hooked up, but you can at least see where I'm going with the new engine system.

**Comment by gendeld** [comment]
Are there no plans to remap google search movie results? Google is the only one i've seen that shows times for pretty much every country...

**Comment by erunion** [comment]
The goal with the engine system is so that no matter the feed you're using, you get back the same format of data. 

The international data problem I haven't really found an easy solution for yet, which has brought me to start work on this engine system so if someone else does come by at some point, it can be easily adapted for.

Google movie search results work in the browser, but after 5 hours of testing I did in November, I found that, in an automated environment, was unpredictable and unparseable.

**Comment by btamada** [comment]
Anyone know of an alternative to this npm package?

**Comment by i8wu** [comment]
Maybe you can try hooking [Fandango](https://developer.fandango.com/) into this module? Haven't looked into it much though.

**Comment by erunion** [comment]
I started hooking up a Fandango engine, but just haven't had time to finish it up. 

https://github.com/erunion/showtimes/pull/51
